src/Filter/FilterComposite.php

```diff
@@ -122,35 +122,46 @@ public function removeFilter(string $name) : void
      * the "and condition" returns true. Otherwise false
      *
      * @param string $property Parameter will be e.g. Parent\Namespace\Class::method
+     * @param null|object $instance Instance on which to perform reflection;
+     *     used primarily with anonymous objects.
      */
-    public function filter(string $property) : bool
+    public function filter(string $property, $instance = null) : bool
```

Comment:
If I remember correctly, we consider this change as a BC Break.

In case of extension we will get warning:
```
Warning: Declaration of ... should be compatible with ...
```

and as of PHP 8:
```
Fatal error: Declaration of ... must be compatible with ...
```

It is the same situation as we have here:
https://github.com/laminas/laminas-paginator/pull/13#discussion_r479628827

Reply:
Bad habits die hard. :frowning_face: 

I've retargeted to 4.0, and will update the interfaces and mark classes final once I've created the 4.0.x branch.